### PR TITLE
Openldap with mysql

### PR DESCRIPTION
**Pull request by mattiashem**
Here is the openldap with mysql backend. It setup to use a local install mysql. But can be plugged out later.

Should I add mysql replicate ?

### CODE REVIEW

File: bin/public/installLdapmysql.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    if os.path.exists("/etc/openldap/slapd.conf"):
+
+      print("There is a slapd-config file will not install")
+
+    else:
+      '''
+      No configfile move on
+      '''
+      password =  generate_password(length=16,chars=string.letters + string.digits)
+      x('yum install wget make gcc  mysql-devel unixODBC-devel groff -y')
```
Comment (kribor) [inline bin/public/installLdapmysql.py]: Don't forget to remove compilers etc. after the installation...

File: bin/public/installLdapmysql.py [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+      x('yum install mysql-connector-odbc -y')
+      x('odbcinst -j')
+      x('\cp -f {0}var/ldap/odbc.ini /etc/odbc.ini'.format(app.SYCO_PATH))
+      x("sed -i 's/SQL_PASS/{0}/g' /etc/odbc.ini".format(password))
+      x('\cp -f {0}var/ldap/init-sql.sql /tmp/init-sql.sql'.format(app.SYCO_PATH))
+      x("sed -i 's/SQL_PASS/{0}/g' /tmp/init-sql.sql".format(password))
+      x('yum install mysql-server -y')
+      x('service mysqld restart')
+      x('mysql < /tmp/init-sql.sql')
+      #x("rm -rf /tmp/init-sql.sql")
```
Comment (kribor) [inline bin/public/installLdapmysql.py]: Please choose whether you want to remove it or not... Either remove the line or remove the comment ;)

File: bin/public/installLdapmysql.py [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+      No configfile move on
+      '''
+      password =  generate_password(length=16,chars=string.letters + string.digits)
+      x('yum install wget make gcc  mysql-devel unixODBC-devel groff -y')
+      os.chdir('/opt')
+      x('wget ftp://mirror.switch.ch/mirror/OpenLDAP/openldap-release/openldap-2.4.43.tgz')
+      x('tar zxvf openldap-*.tgz')
+      x('rm -rf  openldap-*.tgz')
+      x('mv openldap-* openldap')
+      os.chdir('/opt/openldap')
```
Comment (arlukin) [inline bin/public/installLdapmysql.py]: You can use x() here too. We have a cwd option to x, that sets current working directory. and the above os.chdir shouldnt be needed.

File: bin/public/installLdapmysql.py [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+      print("There is a slapd-config file will not install")
+
+    else:
+      '''
+      No configfile move on
+      '''
+      password =  generate_password(length=16,chars=string.letters + string.digits)
+      x('yum install wget make gcc  mysql-devel unixODBC-devel groff -y')
+      os.chdir('/opt')
+      x('wget ftp://mirror.switch.ch/mirror/OpenLDAP/openldap-release/openldap-2.4.43.tgz')
```
Comment (arlukin) [inline bin/public/installLdapmysql.py]: you should use the install.download_file (might have another name, you find it in other install scripts)... 
Then the script will download the tgz script into a temprary folder that are automatically deleteted.

File: bin/public/installLdapmysql.py [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+
+      
+      #Install mysql 
+      x('yum install mysql-connector-odbc -y')
+      x('odbcinst -j')
+      x('\cp -f {0}var/ldap/odbc.ini /etc/odbc.ini'.format(app.SYCO_PATH))
+      x("sed -i 's/SQL_PASS/{0}/g' /etc/odbc.ini".format(password))
+      x('\cp -f {0}var/ldap/init-sql.sql /tmp/init-sql.sql'.format(app.SYCO_PATH))
+      x("sed -i 's/SQL_PASS/{0}/g' /tmp/init-sql.sql".format(password))
+      x('yum install mysql-server -y')
```
Comment (arlukin) [inline bin/public/installLdapmysql.py]: I think you should use the standard mysql installation. And than adapt this installation to that. 
"syco install-mysql " you know.. Than we get replication and others automatially.

syco install-mysql should be executed before syco install-ldapmysql

**Comment by arlukin** [comment]
We had some references to syco-private in the code, so I moved all the code to syco-private instead.

**Pull request closed without merging**